apps/system/js/notifications.js

```diff
@@ -122,6 +126,86 @@ var NotificationScreen = {
     }, function(err) {
       console.error('VersionHelper failed to lookup version settings.');
     });
+
+
+  },
+
+  getStore: function ns_getStore() {
+    return new Promise(resolve => {
+      if (this._storeRef) {
+        return resolve(this._storeRef);
+      }
+      navigator.getDataStores(this.STORE_NAME).then(stores => {
+        this._storeRef = stores[0];
+        return resolve(this._storeRef);
+      });
+    });
+  },
+
+  increaseAppCounter: function ns_udpateCounter(detail) {
+    this.debug('increaseAppCounter('+JSON.stringify(detail)+')');
+    this.debug('has url = ' + this._countersPerApp.hasOwnProperty(detail.manifestURL));
+    this._countersPerApp.hasOwnProperty(detail.manifestURL) ?
+      this._countersPerApp[detail.manifestURL] += 1 :
+      this._countersPerApp[detail.manifestURL] = 1;
+
+    function addCounter(store, name, count) {
+      store.add(count, name).then( id => {
+        this.debug('stored ' + id);
+      });
+    };
+
+    function updCounter(store, name, count) {
+      store.put(count, name).then( id => {
+        this.debug('updated ' + id);
+      });
+    };
+
+    this.getStore().then( store => {
+      store.getLength().then( storeLength => {
+        if(storeLength == 0) {
+          this.debug('empty store');
+          Object.getOwnPropertyNames(this._countersPerApp).forEach( item => {
+            addCounter.call(this, store, item, this._countersPerApp[item]);
+          });
+        } else {
+          this.debug('non empty store');
+          updCounter.call(this, store, detail.manifestURL, this._countersPerApp[detail.manifestURL]);
+        }
+      });
+    });
+  },
+
+  decreaseAppCounter: function ns_decAppCounter(appURL) {
+    // very rough implementation - easy to get negative count.
```

Comment:
actually it's an old comment which needs to be removed, since current implementation won't save values ```<= 0``` to the storage. But here is one issue which probably shall be fixed - local counter ```this._countersPerApp[appURL]``` still could have value ```<= 0```